tests/wpt/mozilla/tests/mozilla/Channel_postMessage_with_second_transfer_in_timeout.window.js

```diff
@@ -22,6 +22,14 @@ async_test(function(t) {
       t.step_timeout(function () {
         // Transfer the port to the iframe.
         TARGET.postMessage("ports", "*", evt.ports);
+
+        // Keep the event-loop busy for a while,
+        // which will intermittently result in the iframe
+        // starting the "complete port transfer" flow,
+        // before the window global could finish it's own.
+        for (var i = 0; i < 99999; i++) {
```

Comment:
Should we use Date.now() and block for a specific amount of time instead? Alternatively, a synchronous XHR that delays the response by a certain amount of time.

Reply:
I think it might be better to keep it loose, because the unknown is how long it takes for the iframe to start the workflow with the constellation, which is going to vary(it's basically the IPC). So I wouldn't know exactly for how long to block the event-loop.

Additionally, the test will cover all branches in the code, albeit not within the same run, due to the intermittent nature of whether the iframe requests the completion of the transfer, or not, by the time the event-loop is unblocked(and the test shold pass either way).

Reply:
Ok I'm trying to write two tests instead, and it seems it's intermittent for another reason, maybe a bug. Looking into it...

Reply:
Ah ok I get it, the timer doesn't actually go through the task queue, it's fired directly when receiving a message from the scheduler, so sometimes it "jumps ahead" of the task queued to "maybe add the port", and then the port is transferred into the iframe before the global even had a chance to "add it" and request completion from the constellation. 

The goal of the test is for the "maybe add pending ports" task to run first, so the global requests completion, and then run the timer which transfers out the port before the global has received the buffer from the constellation, hence branching into the "transfer failed" stuff. 

But when the timer runs first, then the port is simply transferred and the global will not try to complete the transfer since the port is gone.

Reply:
Ok so I think it's actually pretty hard to write a test that will be deterministic, so there are in fact three possibilities:
1. The timer fires first, port goes straight to the iframe.
2. The "maybe add pending port" task runs first, starts the complete transfer flow, then the timer runs, port is transferred, then the global returns the buffer, then the iframe requests transfer completion.
3. Same as 2, except the iframe requests transfer completion before the other global returned the buffer. 

Maybe we can just keep one test, since it passes in all cases, despite it being rather fuzzy?

Reply:
The reason I ask about using Date.now() is that blocking the event loop for a specific amount of time is easier to reason about than 9999 iterations of an operation like JSON.parse. How does a time-based loop cause different results?

Reply:
Ok, I switched to using a sync XHR.